docarray/array/storage/weaviate/backend.py

```diff
@@ -0,0 +1,290 @@
+        """
+        return ''.join([i for i in uuid.uuid1().hex if not i.isdigit()]).capitalize()
+
+    def _get_schema_by_name(self, cls_name: str) -> Dict:
+        """Return the schema dictionary object with the class name
+        Content of the all dictionaries by this method are the same except the name
+        of the weaviate's ``class``
+
+        :param cls_name: the name of the schema/class in weaviate
+        :return: the schema dictionary
+        """
+        # TODO: ideally we should only use one schema. this will allow us to deal with
+        # consistency better
+        return {
+            'classes': [
+                {
+                    'class': cls_name,
+                    "vectorizer": "none",
+                    # TODO: this skips checking embedding dimension but might not
+                    # work if want to leverage weaviate for vector search
+                    'vectorIndexConfig': {'skip': True},
+                    'properties': [
+                        {
+                            'dataType': ['blob'],
+                            'name': '_serialized',
+                            'indexInverted': False,
+                        },
+                    ],
+                },
+                {
+                    'class': cls_name + 'Meta',
+                    "vectorizer": "none",
+                    'vectorIndexConfig': {'skip': True},
+                    'properties': [
+                        {
+                            'dataType': ['string[]'],
+                            'name': '_offset2ids',
+                            'indexInverted': False,
+                        },
+                    ],
+                },
+            ]
+        }
+
+    def _load_or_create_weaviate_schema(self, cls_name: Optional[str] = None):
+        """Create a new weaviate schema for this :class:`DocumentArrayWeaviate` object
+        if not present in weaviate or if ``cls_name`` not provided, else if ``cls_name`` is provided
+        load the object with the given ``cls_name``
+
+        :param cls_name: if provided, load this :class:`DocumentArrayWeaviate` object with
+            the data stored in weaviate. If ``cls_name`` is ``None``, the create a schema in weaviate
+            with a newly generated class name.
+        :return: the schemas of this :class`DocumentArrayWeaviate` object and its meta
+        """
+        if not cls_name:
+            doc_schemas = self._get_schema_by_name(self._get_weaviate_class_name())
+            while self._client.schema.contains(doc_schemas):
+                doc_schemas = self._get_schema_by_name(self._get_weaviate_class_name())
+            self._client.schema.create(doc_schemas)
+            return doc_schemas
+
+        doc_schemas = self._get_schema_by_name(cls_name)
+        if self._client.schema.contains(doc_schemas):
+            return doc_schemas
+
+        self._client.schema.create(doc_schemas)
+        return doc_schemas
```

Comment:
```suggestion
        doc_schemas = self._get_schema_by_name(cls_name or self._get_weaviate_class_name())
        if not self._client.schema.contains(doc_schemas):
            self._client.schema.create(doc_schemas)

        return doc_schemas
```

Reply:
In the case when `cls_name` is not provided (i.e. when user did not specify the class name), as noted in the comment, I believe a new class should be created as we do not want to use previous/existing schema that may contain non-empty data?

Reply:
Yes, I agree. But my suggestions do not change the logic.

Reply:
ah, you are right. forgot my code.